src/Search/SearchServiceProvider.php

```diff
@@ -58,19 +58,27 @@ public function register()
      */
     public function boot()
     {
-        // The rest of these we can resolve in the when->needs->give callback,
-        // but we need to resolve at least one regardless so we know which
-        // searchers we need to register gambits for.
         $fullTextGambits = $this->container->make('flarum.simple_search.fulltext_gambits');
+        $gambits = $this->container->make('flarum.simple_search.gambits');
+
+        $searchersWithGambitsWithoutFulltext = array_diff(array_keys($gambits), array_keys($fullTextGambits));
+
+        if (count($searchersWithGambitsWithoutFulltext)) {
+            $affectedGambits = [];
+            foreach ($searchersWithGambitsWithoutFulltext as $searcher) {
+                $affectedGambits += $gambits[$searcher];
+            }
+
+            throw new \RuntimeException('You cannot add gambits to searchers that do not have fulltext gambits. The following searchers have this issue: '.implode(', ', $affectedGambits));
```

Comment:
I'm worried about this for "no-op" extender calls.

Suppose I have an extension that registers a new searcher.

Another extension has that extension as an optional dependency and inject an additional gambit to it.

Now if the original extension is not present, an error will be thrown instead of the extender just having no effect, all for a searcher that just won't be used anyway.

Maybe add a check for `class_exists($searcherClass)`, or do this test inside of the `give()` binding? It will just require merging the two searcher keys from the two array so that all are looped and bound.